### PR TITLE
Add wrapper API for AeroGear Push

### PR DESCRIPTION
**Pull request by danielpassos**
See: [FHMOBSDK-70](https://issues.jboss.org/browse/FHMOBSDK-70)

### CODE REVIEW

**Comment by danielpassos** [comment]
@wei-lee @secondsun @matzew Could you give it a spin?

**Comment by corinnekrych** [comment]
@danielpassos I'm :eyes:  at it too to get make sure ios/android sdk got similar implementation.

Why is Travis unhappy with your last commit?

File: fh-android-sdk/src/main/java/com/feedhenry/sdk/FH.java [hunk 1]
```diff
@@ -418,10 +424,33 @@ public static int getLogLevel() {
...
+
+    /**
+     *
+     * Registers a device to a push network.
+     *
+     * @param pContext  your application's context
+     * @param pCallback the pCallback function to be executed after the device registration is finished
+     */
+    public void pushRegister(final Context pContext, final FHActCallback pCallback) {
+        RegistrarManager.config("fh", AeroGearGCMPushConfiguration.class)
```
Comment (wei-lee) [inline fh-android-sdk/src/main/java/com/feedhenry/sdk/FH.java]: Does this load the push variants from a file? Probably worth adding it to the java docs to make it clear.

**Comment by danielpassos** [comment]
@wei-lee Nopz, I forgot to change that to the JSON loader configuration. Let me update it.

File: fh-android-sdk/src/main/java/com/feedhenry/sdk/FH.java (hunk 1 above)
Comment (danielpassos) [inline fh-android-sdk/src/main/java/com/feedhenry/sdk/FH.java]: @wei-lee Nopz, I forgot to change that to the JSON loader configuration. Let me update it.

**Comment by danielpassos** [comment]
I think it can be the first step to support push as a wrapper. Anyway we need to figure out the best way to let the developer set the other information like `alias`, `categories`, etc.

PS: It's using a AGDroid snapshot for now, maybe merge it in a new `push` branch instead of master for now

File: fh-android-sdk/src/main/java/com/feedhenry/sdk/FH.java [hunk 2]
```diff
@@ -418,10 +425,46 @@ public static int getLogLevel() {
...
+     *     "senderID": "<senderID e.g Google Project ID only for android>",
+     *     "variantID": "<variantID e.g. 1234456-234320>",
+     *     "variantSecret": "<variantSecret e.g. 1234456-234320>"
+     * }
+     * </pre>
+     *
+     * @param pContext  your application's context
+     * @param pCallback the pCallback function to be executed after the device registration is finished
+     */
+    public void pushRegister(final Context pContext, final FHActCallback pCallback) {
```
Comment (matzew) [inline fh-android-sdk/src/main/java/com/feedhenry/sdk/FH.java]: what is triggering the invocation of that method?

Comment (danielpassos) [inline fh-android-sdk/src/main/java/com/feedhenry/sdk/FH.java]: It's a method for developer call after init (for example)

Comment (matzew) [inline fh-android-sdk/src/main/java/com/feedhenry/sdk/FH.java]: should we mention or refernce that ?

Comment (danielpassos) [inline fh-android-sdk/src/main/java/com/feedhenry/sdk/FH.java]: Why not? :)

File: fh-android-sdk/src/main/java/com/feedhenry/sdk/FH.java [hunk 3]
```diff
@@ -418,10 +425,48 @@ public static int getLogLevel() {
...
+        RegistrarManager.config("fh", AeroGearGCMPushJsonConfiguration.class)
+                .loadConfigJson(pContext)
+                .asRegistrar()
+                .register(pContext, new Callback<Void>() {
+                    @Override
+                    public void onSuccess(Void aVoid) {
+                        pCallback.success(new FHResponse(null, null, null, null));
```
Comment (matzew) [inline fh-android-sdk/src/main/java/com/feedhenry/sdk/FH.java]: @wei-lee @danielpassos would it make sense to have at least a JIRA to give the `FHResponse` a slightly better improved ctor for cases like this, where you just pass in a lot of `null`s ?

**Comment by matzew** [comment]
@danielpassos I think the analytics part is missing in here

**Comment by danielpassos** [comment]
@matzew Working on it

**Comment by danielpassos** [comment]
@matzew analytics added

**Comment by danielpassos** [comment]
Now I'm thinking what is the best way to wrap the configs like `alias` and `categories`

**Comment by corinnekrych** [comment]
@danielpassos don't you think we should reuse FH properties file for configuration:
https://github.com/feedhenry/fh-android-sdk/blob/master/example/src/main/assets/fhconfig.properties
For ios we reuse fhconfig.plist, for dotnet @edewit reuses fhconfig.json.

**Comment by danielpassos** [comment]
@corinnekrych Not for now. I'm doing the same of cordova.  `fh.properties` for FeedHenry configurations and `push-config.json` (or whatever name you want) for push configurations

**Comment by danielpassos** [comment]
@corinnekrych @matzew @wei-lee wdyt?

File: fh-android-sdk/src/main/java/com/feedhenry/sdk/FH.java (hunk 3 above)
Comment (wei-lee) [inline fh-android-sdk/src/main/java/com/feedhenry/sdk/FH.java]: +1

**Comment by corinnekrych** [comment]
+1

**Comment by wei-lee** [comment]
Looks good. Just need to get the build pass.

**Comment by danielpassos** [comment]
Build is broke because I'm using aerogear-android-push:jar:2.2.0-SNAPSHOT

I can fix it building the dependency locally in the build but I prefer wait for the release

**Comment by danielpassos** [comment]
@wei-lee Now we are ready to go.

**Pull request merged**